aip/general/0162.md

```diff
@@ -27,124 +27,108 @@ to the version of an API as a whole.
+rpc GetBookRevision(GetBookRevisionRequest) returns (BookRevision) {
+  option (google.api.http) = {
+    get: "/v1/{name=publishers/*/books/*/revisions/*}"
+  };
+}
+
+message GetBookRevisionRequest {
+  // The name of the book revision.
   //   Example: publishers/123/books/les-miserables
   //
   // In order to retrieve a previous revision of the book, also provide
   // the revision ID.
-  //   Example: publishers/123/books/les-miserables@c7cfa2a8
+  //   Example: publishers/123/books/les-miserables/revisions/c7cfa2a8
   string name = 1 [
     (google.api.field_behavior) = REQUIRED,
     (google.api.resource_reference) = {
-      type: "library.googleapis.com/Book"
+      type: "library.googleapis.com/BookRevision"
     }];
 }
 ```
 
-If the user passes a revision ID that does not exist, the API **must** fail
-with a `NOT_FOUND` error.
+If the user passes a revision ID that does not exist, the API must fail with a `NOT_FOUND` error.
 
-APIs **must** return a `name` value corresponding to what the user sent. If the
-user sent a name with no revision ID, the returned `name` string **must not**
-include the revision ID either. Similarly, if the user sent a name with a
-revision ID, the returned `name` string **must** explicitly include it.
+The returned resource revision `should` be a different resource type from the parent resource type.
 
 ### Tagging revisions
 
+There are two types of tags: Server defined and user specified.
+
+Service may support `current`, `latest`, `oldest`, etc as preserved keyword tags.
+
+GET /v1/publishers/{publisher}/books/{book}/revisions/{preserved_keyword}
+
+Note that `publishers/{publisher}/books/{book}/revisions/latest` and `publishers/{publisher}/books/{book}` can differ. The latest revision may differ from the current status of the resource.
+
 APIs implementing resource revisions **may** provide a mechanism for users to
 tag a specific revision with a user provided name by implementing a "Tag
 Revision" custom method:
 
 ```proto
 rpc TagBookRevision(TagBookRevisionRequest) returns (Book) {
   option (google.api.http) = {
-    post: "/v1/{name=publishers/*/books/*}:tagRevision"
+    post: "/v1/{name=publishers/*/books/*/revisions/*}:tagRevision"
     body: "*"
   };
 }
 ```
 
 ```proto
 message TagBookRevisionRequest {
-  // The name of the book to be tagged, including the revision ID.
+  // The name of the book revision to be tagged.
   string name = 1 [
     (google.api.field_behavior) = REQUIRED,
     (google.api.resource_reference) = {
-      type: "library.googleapis.com/Book"
+      type: "library.googleapis.com/BookRevision"
     }];
 
   // The tag to apply.
   // The tag should be at most 40 characters, and match `[a-z][a-z0-9-]{3,38}[a-z0-9]`.
   string tag = 2 [(google.api.field_behavior) = REQUIRED];
 }
 ```
-
-- The `name` field **should** require an explicit revision ID to be provided.
+- The `name` field **should** require an explicit resource revision to be provided.
```

Comment:
I can't comment on undiffed lines, so:

Note that the following is still in the AIP:

```
Once a revision is tagged, the API **must** support using the tag in place of
  the revision ID in `name` fields.
```

so you can do things like `revisions/current` as well as `revisions/{revision_id}`.

Is that desirable? It's basically creating a second resource id in that case. I kind of feel like at that point, the `tag` API just creates a new resource.

Reply:
The tag is more like alias. I am thinking that it doesn't create new resource. Yes, it is a second resource id. revisions/abcd and revisions/myfavorite can point to the same revision resource.

Reply:
I think that's ok? although it does create weird conflicts for ownership.

regardless that provision has been in there so let's just keep it.